lms/config/resources.py

```diff
@@ -17,19 +17,139 @@ class Root:
 
     def __init__(self, request):
         """Return the default root resource object."""
-        self.request = request
+        self._request = request
 
 
 class LTILaunch:
     """Context resource for LTI launch requests."""
 
+    DISPLAY_NAME_MAX_LENGTH = 30
+    """The maximum length of an h display name."""
+
+    GROUP_NAME_MAX_LENGTH = 25
+    """The maximum length of an h group name."""
+
+    USERNAME_MAX_LENGTH = 30
+    """The maximum length of an h username."""
+
     def __init__(self, request):
         """Return the context resource for an LTI launch request."""
-        self.request = request
+        self._request = request
         # This will raise HTTPBadRequest if the request looks like an OAuth
         # redirect request but no DB-stashed LTI params can be found for the
         # request.
-        self.lti_params = lti_params_for(request)
+        self._lti_params = lti_params_for(request)
+        self._authority = self._request.registry.settings["h_authority"]
+
+    @property
+    def h_display_name(self):
```

Comment:
These properties are just the generate functions from util cut and pasted and adapted a little